get-started/part2.md

```diff
@@ -71,15 +71,15 @@ WORKDIR /app
 # Copy the current directory contents into the container at /app
 ADD . /app
 
+# Define environment variable
+ENV NAME World
```

Comment:
I don't think the `$NAME` env-var has to be set before the `RUN`, as this is only used at runtime (when the container is started)

Reply:
What we _should_ change though, is the section about setting a proxy through an `ENV` instruction. That's not best practice, because doing so will bake your proxy configuration into the image, which means that everyone that will try to run your image will also be using the proxy you configured (which is likely unique to your environment).

Instead, we should describe:

- using `--build-arg` to set a proxy during build time
- use the `--env` option at runtime
- refer to the `proxy` configuration in the CLI configuration file

ping @gbarr01

Reply:
A beginner behind a proxy following the guide its going toget  the error because in these guide does not mention about the option for docker build command behind a  proxy.

I think this should be mentioned in some part of guide.
What you think ?

Reply:
Hi, @thaJeztah , 

I'm struggling to understand the use of ARG. I tested its use with the `NAME` variable and it's not working as I expect.  The docs say: _An ARG instruction can optionally include a default value... If an ARG instruction has a default value and if there is no value passed at build-time, the builder uses the default._  I'm not seeing this. 

For example, this code does _not_ display "Moby" but "None". 

*Dockerfile*
`ARG NAME="Moby"`

*Build and Run*
```
docker build --build arg NAME  --tag hellomoby:v1 .
docker run --publish 4000:80 hellomoby:v1
```

When I tweak the build-time to `docker build --build arg NAME="Moby" --tag hellomoby:v2 .`, it's still "None".

Only when I run, `docker run --env NAME="Moby" --publish 4000:80 hellomoby:v2`, does it display as "Moby". And when I remove `ARG` from `Dockerfile` altogether, with the same `run` command, it also displays as "Moby." 

I cannot see how `ARG` is having any effect here. Can you see what I'm going wrong? Seems like using `--env` is good enough.

Reply:
@gbarr01 let me catch up with you IRL to try and explain (probably faster than writing all down)